slam-libraries/viam-cartographer/src/slam_service/config.cc

```diff
@@ -84,6 +84,13 @@ void ParseAndValidateConfigParams(int argc, char** argv,
         throw std::runtime_error("Invalid slam_mode=" + slamService.slam_mode);
     }
 
+    const auto optimize_on_start =
+        ConfigParamParser(FLAGS_config_param, "optimize_on_start=");
+    if (optimize_on_start == "true") {
+        slamService.optimize_on_start = true;
+        LOG(INFO) << "Optimizing map on start, this may take a few minutes";
```

Comment:
[q] should this log still appear if there is no map to optimize?

Reply:
Very good point. Moved this line into `map_builder.cc` so that it's only called before the optimization actually starts.